src/bun.js/api/bun/socket.zig

```diff
@@ -1677,6 +1677,17 @@ fn NewSocket(comptime ssl: bool) type {
             return ZigString.init(text).toValueGC(globalThis);
         }
 
+        pub fn getRemotePort(
+            this: *This,
+            _: *JSC.JSGlobalObject,
+        ) callconv(.C) JSValue {
+            if (this.detached) {
+                return JSValue.jsUndefined();
+            }
```

Comment:
What happens if I don't define a port?

Reply:
Hi. As far as I know this is a non-issue as this API is not directly exposed to end users and I only invoke it in a place where it guaranteed to exist, and, as you can see from that guard clause, it will return undefined if the socket is detached for some reason. :)